### PR TITLE
Add script for generating an IRI based on JSON-LD data 

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: https://github.com/regen-network/regen-registry/issues/820

Also upgraded a few packages (ts-node, typescript...) since I needed them for the new `iri-gen` package.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] targeted `dev` branch
- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all tests passed
- [ ] submit a PR against `master` branch after this one (and other PRs depending on this one, e.g. on regen-network/regen-web, if applicable) get merged

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: iri-gen/iri-gen.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import * as jsonld from 'jsonld';
```
Comment (blushi) [inline iri-gen/iri-gen.ts:1]: this is the main file to be reviewed

**Comment by wgwz** [comment]
Hi @blushi. One detail I found is this:

```
kyle@lookfar4:~/regen/registry-server$ git diff
diff --git a/lerna.json b/lerna.json
index 5d6c444..02fdd7c 100755
--- a/lerna.json
+++ b/lerna.json
@@ -1,4 +1,4 @@
 {
-  "packages": ["server"],
+  "packages": ["server", "iri-gen"],
   "version": "0.0.0"
 }
```
I think we need to add `iri-gen` to the `lerna.json` packages if the desired result of `yarn test` is to run all the tests for all modules.
```
kyle@lookfar4:~/regen/registry-server$ yarn test
yarn run v1.22.15
$ lerna run test
lerna notice cli v3.22.1
lerna info Executing command in 2 packages: "npm run test"
lerna info run Ran npm script 'test' in 'iri-gen' in 3.3s:

> iri-gen@1.0.0 test
> jest

lerna info run Ran npm script 'test' in 'registry-server' in 4.7s:

> registry-server@0.1.0 test
> jest

lerna success run Ran npm script 'test' in 2 packages in 4.7s:
lerna success - iri-gen
lerna success - registry-server
Done in 5.24s.
```

**Review by clevinson — APPROVED** [review]
Great work @blushi !

Few small comment nits, and an idea for a potential follow-up

**Review by wgwz — APPROVED** [review]

File: iri-gen/iri-gen.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import * as blake from 'blakejs';
+
+const { binary_to_base58 } = require('base58-js');
+const createHash = require('sha256-uint8array').createHash;
+
+const IriPrefixGraph = 1;
+const GraphCanonicalizationAlgorithmURDNA2015 = 1;
+const GraphMerkleTreeUnspecified = 0;
+const DigestAlgorithmBLAKE2b256 = 1;
+const iriVersion0 = 0;
```
[the remark this answers is not in the document]
Comment (blushi) [reply iri-gen/iri-gen.ts:11]: good idea, we should be able to do that for the first four (`iriVersion0` being just a regular constant within the go code), I'll create a follow-up issue

**Pull request merged**